engine/runtime/src/main/scala/org/enso/compiler/pass/resolve/TypeNames.scala

```diff
@@ -44,7 +44,19 @@ case object TypeNames extends IRPass {
       ir.unsafeGetMetadata(BindingAnalysis, "bindings analysis did not run")
     ir.copy(bindings = ir.bindings.map { d =>
       val mapped = d.mapExpressions(resolveExpression(bindingsMap, _))
-      doResolveType(bindingsMap, mapped)
```

Comment:
The changes in this file work, but I am not really sure about the intentions. I was hoping the `d.mapExpressions(resolveExpression ...)` will iterate over `IR.DefinitionArgument`, but it doesn't.

Moreover it is all weird in another way:
- `doResolveType` updates metadata, nothing else - but metadata are "mutable" and can be updated in the `IR` without copying the `IR`
- but there `mapExpression` and that usually does `copy` - we create create identical copy of the `IR` just for fun?
- if we just want to traverse the whole tree - wouldn't it be easier to use `IR.preorder`?

Thanks in advance for your review.

Reply:
I think I got it. `IR.DefinitionArgument` isn't expression - e.g. it is ignored by `mapExpressions`.